watertap/flowsheets/crystallization/Crystallizer_MVR.py

```diff
@@ -0,0 +1,391 @@
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+from pyomo.environ import (
+    ConcreteModel,
+    TerminationCondition,
+)
+from pyomo.environ import (
+    ConcreteModel,
+    value,
+    Constraint,
+    Objective,
+    Var,
+    NonNegativeReals,
+    TransformationFactory,
+    units as pyunits,
+    check_optimal_termination,
+)
+from pyomo.network import Arc
+from idaes.core import FlowsheetBlock
+from watertap.core.solvers import get_solver
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util.initialization import (
+    propagate_state,
+)
+from pyomo.util.check_units import assert_units_consistent
+
+from idaes.core import FlowsheetBlock
+from idaes.core.util.model_statistics import degrees_of_freedom
+
+import idaes.core.util.scaling as iscale
+import idaes.logger as idaeslog
+from watertap.core.solvers import get_solver
+from idaes.core import UnitModelCostingBlock
+
+from watertap.property_models.unit_specific import cryst_prop_pack as props
+from watertap.unit_models.Crystallizer_revised import Crystallization
+from watertap.costing import WaterTAPCosting, CrystallizerCostType
+
+from io import StringIO
+from pyomo.util.infeasible import (
+    log_infeasible_constraints,
+)
+from watertap.unit_models.steam_heater_0D import SteamHeater0D, Mode
+from idaes.models.unit_models import Heater, Separator, Mixer, Product, Feed
+from idaes.models.unit_models.mixer import MomentumMixingType
+from watertap.core.solvers import get_solver
+from watertap.unit_models.mvc.components.lmtd_chen_callback import (
+    delta_temperature_chen_callback,
+)
+from idaes.models.unit_models.heat_exchanger import (
+    HeatExchangerFlowPattern,
+)
+from watertap.unit_models.mvc.components import Compressor
+from watertap.property_models.unit_specific import cryst_prop_pack as props
+import watertap.property_models.water_prop_pack as props_w
+import watertap.property_models.NaCl_T_dep_prop_pack as props_nacl
+from pyomo.common.log import LoggingIntercept
+from idaes.models.unit_models.translator import Translator
+import logging
+from watertap.core.util.model_debug_mode import activate
+activate()
+
+
+
+__author__ = "Elmira Shamlou"
+
+
+def main():
+    solver = get_solver()
+    m = build()
+    set_operating_conditions(m)
+    initialize_system(m, solver=solver)
+
+    #optimize_set_up(m)
+
+    #interval_initializer(m)
+    #solve(m, solver=solver)
+
+    #print("\n***---optimization results---***")
+    #display_system(m)
+    #display_design(m)
+    #display_state(m)
+
+    return m
+
+
+def build():
+
+    # flowsheet set up
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(dynamic=False)
+
+    # properties
+    m.fs.properties_vapor = props_w.WaterParameterBlock()
+    m.fs.properties = props.NaClParameterBlock()
+    m.fs.properties_nacl = props_nacl.NaClParameterBlock()
```

Comment:
Why use two separate NaCl property models? Any reason not to use NaCl with temp dependence by itself?

Reply:
The key difference between the two property packages is that the crystallizer-specific package includes additional solid and vapor phases, which are not present in the other unit models.